RSKIP 271 Update (#295)
rsksmart/RSKIPs: IPs/RSKIP271.md

After receiving a peg-out request, and if no previous request is waiting in the queue, the Bridge contract will wait T=3 hours to create a peg-out transaction. During the waiting time, if more peg-out requests are received, they will be queued so that all of them can be batched in a single peg-out transaction. When the deadline is reached, a peg-out event will be triggered. Under normal usage, the Bridge would create no more than 8 peg-outs events a day. Under a rush hour of peg-outs, the Bridge may need to create more than one peg-out transaction simultaneously per peg-out event to reduce the transaction size, and input count. This peg-out splitting into transactions does not change from the current protocol.

### New Bridge Methods

> - ### Peg-out Fees

Reviewer: ```suggestion
- #### Peg-out Fees
```
Same for the others

Author: Resolved

    If [RSKIP272](https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP272.md) is not activated, then the peg-out fees will be distributed evenly between all peg-out requests as it is today, and no change is required. 

    A new Bridge method `getEstimatedFeesForNextPegOutEvent()`is created so that users can estimate the peg-out fees. This method returns the fees of a peg-out transaction containing (N+2) outputs and 2 inputs, where N is the number of peg-outs requests waiting in the queue. This method will be deprecated if RSKIP272 is activated.

- ### Peg-out Requests Count

>     A new Bridge method `getQueuedPegoutsCount()` is created. This method returns the amount of peg-outs requests waiting in the queue to be released in the next peg-out transaction.

Reviewer: ```suggestion
    A new Bridge method `getQueuedPegoutsCount()` is created. This method returns the amount of peg-out requests waiting in the queue to be released in the next peg-out transaction.
```

Author: Resolved

- ### Peg-out Transaction Creation Block Number

>     A new Bridge method `getNextPegoutCreationBlockNumber()` is created. This method returns the exact block number where the next peg-out transaction is expected to be created. This is calculated by adding the current block number(after a successful pegout) to a defined value of T=360 RSK blocks(~3 hours) i.e nextPegoutCreationBlockNumber = currentBlockNumber + 360.

Reviewer: ```suggestion
    A new Bridge method `getNextPegoutCreationBlockNumber()` is created. This method returns the exact block number where the next peg-out transaction is expected to be created. This is calculated by adding the current block number (after a successful pegout) to a defined value of T=360 RSK blocks (~3 hours).
```

If [RSKIP272](https://github.com/rsksmart/RSKIPs/blob/master/IPs/RSKIP272.md) is not activated, then the peg-out fees will be distributed evenly between all peg-out requests as it is today, and no change is required. 

> A new Bridge method `getEstimatedFeesForNextPegOutEvent()`is created so that users can estimate the peg-out fees. This method returns the fees of a peg-out transaction containing (N+2) outputs and 2 inputs, where N is the number of peg-outs requests waiting in the queue. This method will be deprecated if RSKIP272 is activated.

Reviewer: Maybe create a new section here, something like 
```
### New Bridge methods
```

Author: Done

### Peg-out Requests Count

> A new Bridge method `getQueuedPegoutsCount()` is created. This method returns the amount of peg-outs requests waiting in the queue to be released in the next peg-out transaction i.e. releaseRequestQueue size.

Reviewer: ```suggestion
A new Bridge method `getQueuedPegoutsCount()` is created. This method returns the amount of peg-outs requests waiting in the queue to be released in the next peg-out transaction.
```
I wouldn't mention anything related to the code

Author: Resolved

### Peg-out Transaction Creation Block Number

> A new Bridge method `getNextPegoutCreationBlockNumber()` is created. This method returns the exact block number where the next peg-out transaction is expected to be created. This is calculated by adding the current block number(after a successful pegout) to a defined value of T=3 hours of RSK blocks i.e nextPegoutCreationBlockNumber = currentBlockNumber + 3 hours of RSK blocks.

Reviewer: I think it's best if we mention the amount of RSK blocks instead of the 3 hours (since that is relative). You could add that it is approximately 3 hours given the current block mining time

Author: Done

> ### RSK Transaction Hash

Reviewer: ```suggestion
### Change in `release_requested` event
```

> Currently the rskTxHash for each peg-out request is used to emit `release_requested`. An update for processing peg-out requests in batch would now use the rskTxHash of the transaction that calls updateCollections(at the time the next pegout height is reached) to emit `release_requested` event.

Reviewer: ```suggestion
Currently, the RSK transaction hash for each peg-out request is used to emit `release_requested` event. An update for processing peg-out requests in batch would now use the RSK transaction hash of the transaction that calls updateCollections (at the time the next peg-out height is reached and a batch peg-out transaction is created) to emit `release_requested` event.
```

Author: Done
